stub/src/main/java/io/grpc/stub/AbstractStub.java

```diff
@@ -189,4 +189,20 @@ public final S withCallCredentials(CallCredentials credentials) {
   public final S withWaitForReady() {
     return build(channel, callOptions.withWaitForReady());
   }
+
+  /**
+   * Returns a new stub that limits the maximum acceptable message size from a remote peer.
+   */
+  @ExperimentalApi("https://github.com/grpc/grpc-java/issues/2563")
+  public final S withMaxInboundMessageSize(int maxSize) {
+    return build(channel, callOptions.withMaxInboundMessageSize(maxSize));
+  }
+
+  /**
+   * Returns a new stub that limits the maximum acceptable message size to send a remote peer.
```

Comment:
Then `CallOptions.withMaxOutboundMessageSize()` shouldn't refer to ManagedChannel either.